src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -46,71 +49,128 @@ internal class WorkloadUpdateCommand : InstallingWorkloadCommand
-                    throw new GracefulException(string.Format(LocalizableStrings.WorkloadCacheDownloadFailed, e.Message), e, isUserError: false);
+                    var workloadHistoryRecords = _workloadInstaller.GetWorkloadHistoryRecords().OrderBy(r => r.TimeStarted).ToList();
+                    if (!int.TryParse(_fromHistorySpecified, out int index) || index < 1 || index > workloadHistoryRecords.Count)
+                    {
+                        throw new GracefulException(LocalizableStrings.WorkloadHistoryRecordNonIntegerId, isUserError: true);
+                    }
+
+                    _workloadHistoryState = workloadHistoryRecords[index - 1];
+                    return _workloadHistoryState;
                 }
-            }
-            else if (_printDownloadLinkOnly)
-            {
-                var packageUrls = GetUpdatablePackageUrlsAsync(_includePreviews).GetAwaiter().GetResult();
-                Reporter.WriteLine("==allPackageLinksJsonOutputStart==");
-                Reporter.WriteLine(JsonSerializer.Serialize(packageUrls, new JsonSerializerOptions() { WriteIndented = true }));
-                Reporter.WriteLine("==allPackageLinksJsonOutputEnd==");
-            }
-            else if (_adManifestOnlyOption)
-            {
-                _workloadManifestUpdater.UpdateAdvertisingManifestsAsync(_includePreviews, string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption)).Wait();
-                Reporter.WriteLine();
-                Reporter.WriteLine(LocalizableStrings.WorkloadUpdateAdManifestsSucceeded);
-            }
-            else if (_printRollbackDefinitionOnly)
-            {
-                var workloadSet = WorkloadSet.FromManifests(_workloadResolver.GetInstalledManifests());
 
-                Reporter.WriteLine("==workloadRollbackDefinitionJsonOutputStart==");
-                Reporter.WriteLine(workloadSet.ToJson());
-                Reporter.WriteLine("==workloadRollbackDefinitionJsonOutputEnd==");
+                return null;
             }
-            else
+        }
+
+        public override int Execute()
+        {
+            WorkloadHistoryRecorder recorder = new WorkloadHistoryRecorder(_workloadResolver, _workloadInstaller);
+            recorder.HistoryRecord.CommandName = "update";
+
+            try
             {
-                try
+                if (!string.IsNullOrWhiteSpace(_downloadToCacheOption))
                 {
-                    UpdateWorkloads(_includePreviews, string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption));
+                    try
+                    {
+                        var workloadIds = GetUpdatableWorkloads();
+                        recorder.HistoryRecord.WorkloadArguments = workloadIds.Select(id => id.ToString()).ToList();
+                        recorder.Run(() => DownloadToOfflineCacheAsync(new DirectoryPath(_downloadToCacheOption), _includePreviews, workloadIds).Wait());
```

Comment:
Downloading to history shouldn't change the current versions or workloads that are installed, so we shouldn't need to create a workload history for it.